File: leetcode challenges/Roman to Integer/roman_to_integer.py
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        
        numbers = []
        total = 0

        for char in s:
            if char == "I":
                numbers.append(1)
            elif char == "V":
                numbers.append(5)
            elif char == "X":
                numbers.append(10)
            elif char == "L":
                numbers.append(50)
            elif char == "C":
                numbers.append(100)
            elif char == "D":
                numbers.append(500)
            elif char == "M":
                numbers.append(1000)
        
        for i in range(len(numbers)-1):
            if numbers[i] < numbers[i+1]:
                total -= numbers[i]
            else:
                total += numbers[i]
        
        total += numbers[len(numbers)-1]

        return total

File: leetcode challenges/Roman to Integer/romanToInteger.cpp
class Solution {
public:
    int romanToInt(string s) {

        vector<int> numbers;
        int total = 0;

        for(int i=0; i<s.size(); i++){
            switch(s.at(i)){
                case 'I':
                    numbers.push_back(1);
                    break;
                case 'V':
                    numbers.push_back(5);
                    break;
                case 'X':
                    numbers.push_back(10);
                    break;
                case 'L':
                    numbers.push_back(50);
                    break;
                case 'C':
                    numbers.push_back(100);
                    break;
                case 'D':
                    numbers.push_back(500);
                    break;
                case 'M':
                    numbers.push_back(1000);
                    break;
            }
        }

        for(int i=0; i<numbers.size()-1; i++){
            if(numbers.at(i) < numbers.at(i+1)) total -= numbers.at(i);
            else total += numbers.at(i);
        }

        total += numbers.at(numbers.size()-1);
        return total;
    }
};
